ldap/servers/plugins/sync/sync_refresh.c

```diff
@@ -705,13 +815,14 @@ sync_send_entry_from_changelog(Slapi_PBlock *pb, int chg_req __attribute__((unus
     slapi_pblock_get(search_pb, SLAPI_PLUGIN_INTOP_RESULT, &rv);
     if (rv == LDAP_SUCCESS) {
         slapi_pblock_get(search_pb, SLAPI_PLUGIN_INTOP_SEARCH_ENTRIES, &entries);
-        if (entries)
+        if (entries) {
             db_entry = *entries; /* there can only be one */
+        }
     }
 
     if (db_entry && sync_is_entry_in_scope(pb, db_entry)) {
         LDAPControl **ctrl = (LDAPControl **)slapi_ch_calloc(2, sizeof(LDAPControl *));
-        sync_create_state_control(db_entry, &ctrl[0], chg_type, NULL);
+        sync_create_state_control(db_entry, &ctrl[0], chg_type, NULL, cookie->openldap_compat);
```

Comment:
I'm trying to understand this comment, but maybe I can't quite see what you are thinking of. Are you saying that if someone were to LDAPMOD and del the entryuuid attr that would cause a problem?

Reply:
Sorry for this not clear remark.

sync_create_state_control can fails (if the db entry has no SLAPI_ATTR_ENTRYUUID). In such case we should not send the entry. So the suggestion is to test the returned code of sync_create_state_control to skip the entry.

Reply:
I'm not sure about this. If the entry has lost it's entryUUID, then that would indicate db corruption or tampering, so the  `sync_create_state_control` should fail - and does, with "sync_create_state_control - Some entries are missing entryUUID. Unable to proceed. You may need to re-run the entryuuid fixup". If the entry *had* a entryUUID and it was removed, this also indicates tampering so the error is valid. It's worth noting the syntax of entryuuid is: `SYNTAX 1.3.6.1.1.16.1 SINGLE-VALUE NO-USER-MODIFICATION USAGE directoryOperation` which prevents user modification of the value, and prevents users *removing* the value. So if the entryUUID goes missing, then this is an error and we SHOULD fail the whole sync repl operation for openldap because it means we could potentially replicate and corrupt data silently. So this is the correct solution to fail fast and hard to prevent replicating corrupted data (especially as openldap's replication processing for incoming data is very fragile).

Reply:
I agree, if it detects that entryUUID is missing (so sync_create_state_control fails) it should fails. This is what I wanted to say in my initial comment.
The decision to just skip this update or to stop the full sync_repl search is up to you. I think it makes sense to break the search as the missing entryUUID indicates 389-DS is not longer able top sync with openldap.

Reply:
Ahhhhhh I see what you are saying now! Because the `rc` of `sync_create_state_control` isn't checked the entry is skipped rather than the search terminated. So I think here, we should have an `rc = sync_create_state_control(...)` and then if `!= 0`, we return the rc similar to `sync_srch_refresh_pre_entry`

Reply:
Add rc check :) thank you!